ui/packages/consul-ui/app/services/repository/binding-rule.js

```diff
@@ -0,0 +1,37 @@
+import RepositoryService from 'consul-ui/services/repository';
+import statusFactory from 'consul-ui/utils/acls-status';
+import isValidServerErrorFactory from 'consul-ui/utils/http/acl/is-valid-server-error';
+import { PRIMARY_KEY, SLUG_KEY } from 'consul-ui/models/binding-rule';
+import dataSource from 'consul-ui/decorators/data-source';
+
+const isValidServerError = isValidServerErrorFactory();
+const status = statusFactory(isValidServerError, Promise);
+const MODEL_NAME = 'binding-rule';
+
+export default class BindingRuleService extends RepositoryService {
+  getModelName() {
+    return MODEL_NAME;
+  }
+
+  getPrimaryKey() {
+    return PRIMARY_KEY;
+  }
+
+  getSlugKey() {
+    return SLUG_KEY;
+  }
+
+  @dataSource('/:ns/:dc/binding-rules')
+  async findAllByDatacenter() {
+    return super.findAllByDatacenter(...arguments);
+  }
+
+  @dataSource('/:ns/:dc/binding-rule/:id')
+  async findBySlug() {
+    return super.findBySlug(...arguments);
+  }
```

Comment:
Did we say we were gonna remove this if it's not used?

Reply:
yes, I missed this one. Removed.